### ISSUE TITLE
[ stable/concourse ] Using a real BTRFS partition make initContainer fail/slow

### ISSUE THREAD

**Issue by Typositoire** [helm/charts#13284]
FEATURE REQUEST

When using a real BTRFS partition (not a loop mount) the current init container doesn't work.
![image](https://user-images.githubusercontent.com/721017/56740103-3e28b800-673e-11e9-8b49-488dac31e70a.png)

It is possible to either account for this OR let us overwrite the current initContainer script ?

**Comment by Typositoire** [comment]
you need to do `btrfs subvolume delete <volume>`

**Comment by cirocosta** [comment]
Hey @Typositoire ,

I just added this to our backlog to ensure that we have the proper testing and the functionality around this (seems related to https://github.com/helm/charts/pull/12398), which seems related to https://github.com/helm/charts/pull/12398.

At the moment, we're focusing on a few other areas, but we'll soon get to this one.

Thx for reporting!

**Comment by Typositoire** [comment]
Coll thanks!

**Comment by Typositoire** [comment]
A good interim solution would be to make the initscript a Value ? Would you be down for a PR ?

**Comment by taylorsilva** [comment]
### Dev Notes

We were able to reproduce this issue on GKE by doing the following:

- Created a persistent disk that GKE then formatted with `btrfs`: https://kubernetes.io/docs/concepts/storage/volumes/#creating-a-pd
- Deployed a pod running a concourse worker that we connected to an existing Concourse that we have running. We tagged this worker with `btrfs-test`
- Ran a simple pipeline that would run on our tagged worker, thus creating volumes
- We then deleted the pod and recreated it. This detached and then reattached the same PD. We saw the same logs by ssh'ing onto the node and using `docker logs` to view the logs.
  - At this point, we've reproduced the issue
- We then deleted the pod and recreated it with the changes in this PR: https://github.com/helm/charts/pull/12398 and it worked!

Next Steps:
We need to figure out how to test this and then implement a solution

**Comment by Typositoire** [comment]
In my case I don't use any PVC, it's the node partition directly but the end result should be the same. I have 1-1 node-worker setup, it's easier to maintain :p I can't wait to have #12398 merged, it makes every worker restart a force destroy otherwise the node take 5 years to come back.

**Comment by taylorsilva** [comment]
### Dev Notes

We created a test to cover this use-case. Just need the previous PR mentioned to be merged into the chart.

**Comment by cirocosta** [comment]
Testing done - waiting on:

- [ ] merge of https://github.com/helm/charts/pull/14941, then
- [ ] merge of https://github.com/concourse/concourse/pull/3997

Thanks!
